kolibri/core/discovery/models.py

```diff
@@ -117,7 +120,16 @@ class Meta:
         proxy = True
 
     def save(self, *args, **kwargs):
+        from kolibri.core.discovery.utils.network.connections import (
+            capture_network_state,
+        )
+        from kolibri.core.discovery.utils.network.client import NetworkClient
```

Comment:
The two imports are causing circular dependency bugs when imported from the top of the file, so they have been added as part of the function to solve the issue.